0.8.0+ crashes - non default schedule configuration

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

### Terraform CLI and Terraform AWS Provider Version
Terraform 1.8.5
<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* metabase_database

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# you can also create a reproducer repo and share the link here.
```

### Debug Output

<!---
If possible, please provide the debug output as this will make it easier to identify the issue.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->
```
Stack trace from the terraform-provider-metabase_v0.8.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7f26b5]

goroutine 30 [running]:
terraform-provider-metabase/internal/provider.buildScheduleSettings(0x0)
        terraform-provider-metabase/internal/provider/database_resource.go:289 +0x115
terraform-provider-metabase/internal/provider.buildSchedules(0xc00043e750)
        terraform-provider-metabase/internal/provider/database_resource.go:305 +0x165
terraform-provider-metabase/internal/provider.mapDatabaseToState({0xd3f288, 0xc00035b6e0}, 0xc00043e750, 0xc0000003c0)
        terraform-provider-metabase/internal/provider/database_resource.go:281 +0x19c
terraform-provider-metabase/internal/provider.(*DatabaseResource).Read(0xc0003f0120, {0xd3f288, 0xc00035b6e0}, {{{{0xd43eb8, 0xc0004e7590}, {0xb262e0, 0xc0004e6c30}}, {0xd45b70, 0xc0004bad70}}, 0xc0003f0130, ...}, ...)
        terraform-provider-metabase/internal/provider/database_resource.go:108 +0x1bb
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadResource(0xc0000f76c0, {0xd3f288, 0xc00035b6e0}, 0xc00035b740, 0xc00063b6c8)
        github.com/hashicorp/terraform-plugin-framework@v1.8.0/internal/fwserver/server_readresource.go:101 +0x62e
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ReadResource(0xc0000f76c0, {0xd3f288?, 0xc00035b5f0?}, 0xc00038ad00)
        github.com/hashicorp/terraform-plugin-framework@v1.8.0/internal/proto6server/server_readresource.go:55 +0x38e
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ReadResource(0xc0001f08c0, {0xd3f288?, 0xc00035ae70?}, 0xc0000ecf50)
        github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov6/tf6server/server.go:784 +0x309
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ReadResource_Handler({0xbf7000, 0xc0001f08c0}, {0xd3f288, 0xc00035ae70}, 0xc000497480, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:482 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001a5000, {0xd3f288, 0xc00035ad50}, {0xd44820, 0xc000488000}, 0xc0000c38c0, 0xc00027db90, 0x122c528, 0x0)
        google.golang.org/grpc@v1.63.2/server.go:1369 +0xdf8
google.golang.org/grpc.(*Server).handleStream(0xc0001a5000, {0xd44820, 0xc000488000}, 0xc0000c38c0)
        google.golang.org/grpc@v1.63.2/server.go:1780 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.63.2/server.go:1019 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 35
        google.golang.org/grpc@v1.63.2/server.go:1030 +0x125

Error: The terraform-provider-metabase_v0.8.0 plugin crashed!
```
### Panic Output

<!--- If Terraform produced a panic, please provide the output of the `crash.log`. --->

### Expected Behavior

<!--- What should have happened? --->
All should work, as it was working with v 0.7.1 before.

### Actual Behavior

<!--- What actually happened? --->
It crashes during plan phase.

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform plan`

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? API documentation? For example:
--->
Not sure, but we think it's due to a change of schedules behavior in Metabase - https://github.com/bnjns/terraform-provider-metabase/compare/v0.7.1...v0.8.0#diff-48fd8dd51e44a1407938cfa2c28f5920f478fa5bfb58e149b459b3870ef7e62bR287 - in our case we do use non-default settings for schedules and it seems it's broken.

![image](https://github.com/bnjns/terraform-provider-metabase/assets/17637831/020289ba-e196-4533-947b-5d2245880662)

* #0000

--- a/internal/provider/database_resource.go
+++ b/internal/provider/database_resource.go
@@ -285,6 +285,10 @@ func mapDatabaseToState(ctx context.Context, db *database.Database, target *Data
 }
 
 func buildScheduleSettings(settings *database.ScheduleSettings) basetypes.ObjectValue {
+	if settings == nil {
+		return types.ObjectNull(schema.DatabaseScheduleType.AttributeTypes())
+	}
+
 	scheduleSettings, _ := types.ObjectValue(schema.DatabaseScheduleType.AttributeTypes(), map[string]attr.Value{
 		"type":   types.StringValue(string(settings.Type)),
 		"day":    transforms.ToTerraformString((*string)(settings.Day)),

--- a/internal/provider/database_resource_test.go
+++ b/internal/provider/database_resource_test.go
@@ -76,6 +76,25 @@ func TestBuildSchedules(t *testing.T) {
 
 		assert.Equal(t, 2, len(schedules.Attributes()))
 	})
+
+	t.Run("a database with nil schedules should return a map with null values", func(t *testing.T) {
+		db := database.Database{
+			Schedules: &database.Schedules{
+				MetadataSync:     nil,
+				CacheFieldValues: nil,
+			},
+		}
+
+		schedules, diags := buildSchedules(&db)
+		assert.Zero(t, len(diags))
+		assert.False(t, schedules.IsNull())
+		assert.False(t, schedules.IsUnknown())
+
+		assert.Equal(t, 2, len(schedules.Attributes()))
+		for _, schedule := range schedules.Attributes() {
+			assert.True(t, schedule.IsNull())
+		}
+	})
 }
 
 func TestAccDatabaseResource_PostgreSQL(t *testing.T) {

